radicale/__init__.py

```diff
@@ -425,6 +433,13 @@ def response(status, headers=(), answer=None):
                     environ, base_prefix, path, user)
             except socket.timeout:
                 return response(*REQUEST_TIMEOUT)
+            except vobject.base.VObjectError as e:
+                return response(*UNPROCESSABLE_ENTITY(e))
```

Comment:
> From the RFC:

>>  syntax of the request entity is correct

> This also catches syntax errors.

Indeed. But RFC 4918 continues with: "For example, this error condition may occur if an XML request body contains well-formed (i.e., syntactically correct), but semantically erroneous, XML instructions." Now replace XML by vObject and this is our case.

[Wikipedia](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors) says: "422: The request was well-formed but was unable to be followed due to semantic errors".

There is also a [longer discussion about this at stackoverflow](https://stackoverflow.com/questions/16133923/400-vs-422-response-to-post-of-data) which concludes: "Given the choice between using an error code in the original standard that explicitly doesn't cover the situation, and one from an extension that describes the situation exactly, **I would choose the latter**." (Emphasis by me)

So IMHO 422 is more appropriate then 440, since as I understand it, 422 is about the data which is transferred, while 400 is about he request. Otherwise there would have been no reason for introducing 422 for WebDAV.

> But the 4xx errors are indicating client errors. If a vobject from the storage can't be parsed, it's not a client error.

Well spotted :-) I'll move this code into "PUT" (and related) after we agreed on the error.